web/src/pages/Explore.tsx

```diff
@@ -41,6 +44,42 @@ const Explore = () => {
     });
   }, [location]);
 
+  const { tag: tagQuery, text: textQuery } = query ?? {};
+  const showMemoFilter = Boolean(tagQuery || textQuery);
+
+  const shownMemos = showMemoFilter
+    ? state.memos.filter((memo) => {
+        let shouldShow = true;
+
+        if (tagQuery) {
+          const tagsSet = new Set<string>();
+          for (const t of Array.from(memo.content.match(new RegExp(TAG_REG, "g")) ?? [])) {
+            const tag = t.replace(TAG_REG, "$1").trim();
+            const items = tag.split("/");
+            let temp = "";
+            for (const i of items) {
+              temp += i;
+              tagsSet.add(temp);
+              temp += "/";
+            }
+          }
+          if (!tagsSet.has(tagQuery)) {
+            shouldShow = false;
+          }
+        }
+        return shouldShow;
+      })
+    : state.memos;
+
+  const pinnedMemos = shownMemos.filter((m) => m.pinned);
+  const unpinnedMemos = shownMemos.filter((m) => !m.pinned);
+  const memoSort = (mi: Memo, mj: Memo) => {
+    return mj.displayTs - mi.displayTs;
+  };
+  pinnedMemos.sort(memoSort);
+  unpinnedMemos.sort(memoSort);
```

Comment:
Pin/unpin memo is a personal behaviour and should not be display in explore page.